plugins/catalog-backend-module-incremental-ingestion/src/types.ts

```diff
@@ -75,6 +75,22 @@ export interface IncrementalEntityProvider<TCursor, TContext> {
    * @param burst - a function which performs a series of iterations
    */
   around(burst: (context: TContext) => Promise<void>): Promise<void>;
+
+  /**
+   * If present, this method accepts an incoming event, and maps the
+   * payload to an object containing a delta mutation.
+   *
+   * If a delta is present, the incremental entity provider will apply
+   * it automatically.
+   */
+  onEvent?: (payload: unknown) => {
```

Comment:
Again, I _really_ think that this should be `params: EventParams` instead of the payload. The other fields in the event params are there for good reason :) You'll quickly run into needing those headers or other things (that are there today or get added in the future).

Reply:
I agree. It took me some iterations to end up at EventParams and originally, I only had the payload, too.
I think it is better to support EventParams and stay more aligned.

Reply:
Will do. It will require some changes to the engine as well, since the engine currently handles the actual event, but those should be minimal.

Reply:
Done.